math/vector_cross_product.cpp

```diff
@@ -0,0 +1,129 @@
+/**
+ * @file
+ *
+ * @brief Calculates the [Cross Product](https://en.wikipedia.org/wiki/Cross_product) and the magnitude of two mathematical 3D vectors.
+ *
+ *
+ * @details Cross Product of two vectors gives a vector.
+ * Direction Ratios of a vector are the numeric parts of the given vector. They are the tree parts of the
+ * vector which determine the magnitude (value) of the vector.
+ * The method of finding a cross product is the same as finding the determinant of an order 3 matrix consisting
+ * of the first row with unit vectors of magnitude 1, the second row with the direction ratios of the
+ * first vector and the third row with the direction ratios of the second vector.
+ * The magnitude of a vector is it's value expressed as a number.
+ * Let the direction ratios of the first vector, P be: a, b, c
+ * Let the direction ratios of the second vector, Q be: x, y, z
+ * Therefore the calculation for the cross product can be arranged as:
+ * P x Q:
+ *  	1	1	1
+ *  	a	b	c
+ *  	x	y	x
+ *
+ * @algorithm The direction ratios (DR) are calculated as follows:
```

Comment:
there is no doxygen tag `@algorithm` and will lead to errors
```suggestion
 * The direction ratios (DR) are calculated as follows:
```

Reply:
Done!